src/main/java/Task.java

```diff
@@ -0,0 +1,35 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public Task markAsDone() {
+        return new Task(this.getDescription(), true);
+    }
+
+    public Task markAsUndone() {
+        return new Task(this.getDescription(), false);
+    }
+
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
```

Comment:
Perhaps the use of `this` keyword can be omitted unless the argument and attribute names are the same? The SE Coding Standard guide specifies this.